core/record/_record/recordReader.js

```diff
@@ -201,15 +199,56 @@ export const getParentCodeAttribute = (survey, parentNode, nodeDef) => record =>
 
 // ====== Keys
 
-export const getEntityKeyNodes = (survey, nodeEntity) => record => {
+export const getEntityKeyNodes = (survey, nodeEntity) => (record) => {
   const nodeDefEntity = SurveyNodeDefs.getNodeDefByUuid(Node.getNodeDefUuid(nodeEntity))(survey)
   const nodeDefKeys = SurveyNodeDefs.getNodeDefKeys(nodeDefEntity)(survey)
 
   return R.pipe(
-    R.map(nodeDefKey => getNodeChildByDefUuid(nodeEntity, NodeDef.getUuid(nodeDefKey))(record)),
-    R.flatten,
+    R.map((nodeDefKey) => getNodeChildByDefUuid(nodeEntity, NodeDef.getUuid(nodeDefKey))(record)),
+    R.flatten
   )(nodeDefKeys)
 }
 
 export const getEntityKeyValues = (survey, nodeEntity) =>
   R.pipe(getEntityKeyNodes(survey, nodeEntity), R.map(Node.getValue))
+
+export const getAttributesUniqueSibling = ({ record, attribute, attributeDef }) => {
+  const parentEntity = getParentNode(attribute)(record)
+  const ancestorEntity = getParentNode(parentEntity)(record)
+  const siblingParentEntities = getNodeChildrenByDefUuid(ancestorEntity, NodeDef.getParentUuid(attributeDef))(record)
+
+  return siblingParentEntities.reduce(
+    (siblingsAcc, siblingEntity) => [
+      ...siblingsAcc,
+      ...getNodeChildrenByDefUuid(siblingEntity, NodeDef.getUuid(attributeDef))(record),
+    ],
+    []
+  )
+}
+
+const _isNodeDefUnique = (nodeDef) => NodeDefValidations.isUnique(NodeDef.getValidations(nodeDef))
+
+export const getAttributesUniqueDependent = ({ survey, record, node }) => {
```

Comment:
do we need to add byId or something like that?

Reply:
yes you are right, a name like getUniqueDependentsByUuid would be better... but we have so many similar cases in the code. The return type will be more clear in a near future using TypeScript.